src/lib/duration/constructor.js

```diff
@@ -1,17 +1,24 @@
 import { normalizeObjectUnits } from '../units/aliases';
 import { getLocale } from '../locale/locales';
+import isDurationValid from './valid.js';
 
 export function Duration (duration) {
-    var normalizedInput = normalizeObjectUnits(duration),
-        years = normalizedInput.year || 0,
-        quarters = normalizedInput.quarter || 0,
-        months = normalizedInput.month || 0,
-        weeks = normalizedInput.week || 0,
-        days = normalizedInput.day || 0,
-        hours = normalizedInput.hour || 0,
-        minutes = normalizedInput.minute || 0,
-        seconds = normalizedInput.second || 0,
-        milliseconds = normalizedInput.millisecond || 0;
+    var normalizedInput = normalizeObjectUnits(duration);
+
+    this._isValid = isDurationValid(normalizedInput);
+    this.isValid = function () {
+        return this._isValid;
+    };
+
+    var years = this._isValid && normalizedInput.year || 0,
```

Comment:
We have a `one var` policy in jshint, so this won't fly. Just don't change the code (leave it as it was) -- if its invalid its safe to exit this function. All other functions check for `isValid`, so its fine if the internals are messed up.